### PR TITLE
Add Multidoc2dial Dataset

### PR DESCRIPTION
**Pull request by sivasankalpp**
This PR adds the MultiDoc2Dial dataset introduced in this [paper](https://arxiv.org/pdf/2109.12595v1.pdf )

### CODE REVIEW

**Comment by sivasankalpp** [comment]
@songfeng cc

**Comment by albertvillanova** [comment]
Hi @sivasankalpp, thanks for your PR.

There was a bug in TensorFlow/Keras. We have made a temporary fix in our master branch. Please, merge master into your PR branch, so that the CI tests pass.

```
git checkout multidoc2dial
git fetch upstream master
git merge upstream/master
```

**Comment by sivasankalpp** [comment]
Hi @albertvillanova, I have merged master into my PR branch. All tests are passing. 
Please take a look when you get a chance, thanks!

File: datasets/multidoc2dial/README.md [hunk 1]
```diff
@@ -0,0 +1,465 @@
+---
+annotations_creators:
+- crowdsourced
+language_creators:
+- crowdsourced
+- expert-generated
+languages:
+- en-US
```
Comment (lhoestq) [inline datasets/multidoc2dial/README.md:8]:
```suggestion
- en
```

File: datasets/multidoc2dial/README.md [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+licenses:
+- apache-2.0
+multilinguality:
+- monolingual
+pretty_name: MultiDoc2Dial
+size_categories:
+- unknown
```
Comment (lhoestq) [inline datasets/multidoc2dial/README.md:15]:
```suggestion
  dialogue_domain:
  - 1K<n<10
  document_domain:
  - n<1K
  multidoc2dial:
  - 10K<n<100K
```

File: datasets/multidoc2dial/README.md [hunk 3]
```diff
@@ -0,0 +1,465 @@
...
+    - `turn_id`: the time order of the turn;
+    - `role`: either "agent" or "user";READ
+    - `da`: dialogue act;
+    - `references`: a list of spans with `id_sp` ,  `label` and `doc_id`. `references` is empty if a turn is for indicating previous user query not answerable or irrelevant to the document. **Note** that labels "*precondition*"/"*solution*" are fuzzy annotations that indicate whether a span is for describing a conditional context or a solution.
+    - `utterance`: the human-generated utterance based on the dialogue scene.
+
+### Data Splits
+
+- Training & dev split for dialogue domain
+- Training split only for document domain
```
Comment (lhoestq) [inline datasets/multidoc2dial/README.md:398]:
```suggestion
- Training, dev and test split for default configuration `multidoc2dial`, with respectively 21451, 4201 and 5 examples,
- Training & dev split for dialogue domain, with 3474 and 661 examples,
- Training split only for document domain, with 488 examples.
```

Comment (lhoestq) [reply datasets/multidoc2dial/README.md:398]: Also is it normal that there are only 5 examples in the test set ?

File: datasets/multidoc2dial/README.md [hunk 4]
```diff
@@ -0,0 +1,465 @@
...
+
+  Each dialogue instance includes the following,
+
+  - `dial_id`: the ID of a dialogue;
+  - `turns`: a list of dialogue turns. Each turn includes,
+    - `turn_id`: the time order of the turn;
+    - `role`: either "agent" or "user";READ
+    - `da`: dialogue act;
+    - `references`: a list of spans with `id_sp` ,  `label` and `doc_id`. `references` is empty if a turn is for indicating previous user query not answerable or irrelevant to the document. **Note** that labels "*precondition*"/"*solution*" are fuzzy annotations that indicate whether a span is for describing a conditional context or a solution.
+    - `utterance`: the human-generated utterance based on the dialogue scene.
```
Comment (lhoestq) [inline datasets/multidoc2dial/README.md:393]: Here we actually expect the description of each field of the processed dataset as defined in the `Features` of the dataset script. However I think you described the fields of the original JSON data here ? Could you describe the fields of the processed dataset here instead ?

File: datasets/multidoc2dial/README.md [hunk 5]
```diff
@@ -0,0 +1,465 @@
...
+### Languages
+
+English
+
+## Dataset Structure
+
+### Data Instances
+
+Sample data instance for `dialogue_domain` :
+
```
Comment (lhoestq) [inline datasets/multidoc2dial/README.md:80]: Same here: we expect to see an example from the processed dataset, not from the raw JSON data.

You can get one example by using `load_dataset` and getting the first sample `dataset["train"][0]`

File: datasets/multidoc2dial/README.md [hunk 6]
```diff
@@ -0,0 +1,465 @@
...
+Creative Commons Attribution 3.0 Unported
+
+### Citation Information
+
+@inproceedings{feng2021multidoc2dial,
+    title={MultiDoc2Dial: Modeling Dialogues Grounded in Multiple Documents},
+    author={Feng, Song and Patel, Siva Sankalp and Wan, Hui and Joshi, Sachindra},
+    booktitle={EMNLP},
+    year={2021}
+}
```
Comment (lhoestq) [inline datasets/multidoc2dial/README.md:461]: add backticks for formatting
````suggestion
```bibtex
@inproceedings{feng2021multidoc2dial,
    title={MultiDoc2Dial: Modeling Dialogues Grounded in Multiple Documents},
    author={Feng, Song and Patel, Siva Sankalp and Wan, Hui and Joshi, Sachindra},
    booktitle={EMNLP},
    year={2021}
}
```
````

File: datasets/multidoc2dial/README.md [hunk 7]
```diff
@@ -0,0 +1,465 @@
...
+
+### Contributions
+
+[@songfeng](https://github.com/songfeng) and [@sivasankalpp](https://github.com/sivasankalpp) added this dataset.
```
Comment (lhoestq) [inline datasets/multidoc2dial/README.md:465]:
```suggestion
Thanks to [@songfeng](https://github.com/songfeng) and [@sivasankalpp](https://github.com/sivasankalpp) for adding this dataset.
```

File: datasets/multidoc2dial/README.md [hunk 8]
```diff
@@ -0,0 +1,465 @@
...
+### Dataset Summary
+
+MultiDoc2Dial is a new task and dataset on modeling goal-oriented dialogues grounded in multiple documents. 
+Most previous works treat document-grounded dialogue modeling as a machine reading comprehension task based on a
+single given document or passage. We aim to address more realistic scenarios where a goal-oriented information-seeking
+conversation involves multiple topics, and hence is grounded on different documents. 
+
+### Supported Tasks and Leaderboards
+
+> Supported Task:
```
Comment (lhoestq) [inline datasets/multidoc2dial/README.md:67]: Maybe add a description of the task here ?

File: datasets/multidoc2dial/README.md [hunk 9]
```diff
@@ -0,0 +1,465 @@
...
+- monolingual
+pretty_name: MultiDoc2Dial
+size_categories:
+- unknown
+source_datasets:
+- extended|doc2dial
+task_categories:
+- question-answering
+task_ids:
+- open-domain-qa
```
Comment (lhoestq) [inline datasets/multidoc2dial/README.md:21]: add paperswithcode_id
```suggestion
- open-domain-qa
paperswithcode_id: multidoc2dial
```

**Review by lhoestq — COMMENTED** [review]
This is awesome ! Thanks for adding this dataset. And good job with the dataset card and the dataset script :)

I just added some minor comments here and there:

File: datasets/multidoc2dial/README.md (hunk 3 above)
Comment (songfeng) [reply datasets/multidoc2dial/README.md:398]: We are withholding the full test split for leaderboard or data competition. We included a dummy test split with only 5 examples for now.

**Comment by sivasankalpp** [comment]
Thanks for your feedback @lhoestq. We addressed your comments in the latest commit. Let us know if everything looks okay :)

**Review by lhoestq — APPROVED** [review]
Cool thanks a lot ! LGTM :)
